sdk/core/Azure.Core/src/Shared/ChangeTrackingDictionary.cs

```diff
@@ -17,24 +17,10 @@ public ChangeTrackingDictionary()
         {
```

Comment:
I had made some changes to these types in https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/core/System.ClientModel/tests/client/ClientShared, where

- ChangeTrackingDictionary -> OptionalDictionary
- ChangeTrackingList -> OptionalList
- Optional -> OptionalProperty

They were to enable nullability annotations in ClientModel-based clients.  It might be worth checking that we're keeping these things in sync, because the intention was that the types from ClientShared would be what the generator emits for serialization of unbranded clients, or at least starting there as a starting point we iterate from. Eventually, we want these classes to converge, and that work is tracked as part of https://github.com/Azure/azure-sdk-for-net/issues/41035

Reply:
We are most likely going to stick with Azure.Core implementation for now to avoid introducing too many variables at once.